src/lib.rs

```diff
@@ -15,7 +15,13 @@
 //!  * hash_pubkey_to_point: `BLS12381G2_XMD:SHA-256_SSWU_RO_` with the ASCII-encoded domain separation tag `BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_`
 //!
 //! While the Domain Separation Tag (DST) isn't hardcoded in this crate, we are hardcoding the choice of elliptic curve (BLS12-381), hash function (SHA-256), and variant (minimal-pubkey-size).
-use std::ops::{Add, AddAssign};
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
```

Comment:
Is this needed? I think the `use alloc::` statements should work without this

Reply:
actually it is, it's not included in the prelude

Reply:
Interesting. And `alloc` is the only crate that needs this `extern` statement? Other imports don't need it? (e.g. `use ark_bls12_381::g2::Config as G2Config;` will work without `extern crate ark_bls12_381` in no-std envs?)

Reply:
yeah alloc is a special crate, because you don't declare it in your cargo.toml